lib/spark/dsl/extension.ex

```diff
@@ -878,11 +878,37 @@ defmodule Spark.Dsl.Extension do
       end
 
       unless section.top_level? && path == [] do
-        defmacro unquote(section.name)(body) do
+        # Handle mixed syntax - personal_details(opts, do: block)
+        defmacro unquote(section.name)(opts, [do: _] = _block) do
```

Comment:
🤔 what if we actually just could handle both? Do we want to? We could theoretically just call it with just opts and then call it with just `do` and it should "just work"

Reply:
I've pushed up the same handling for mixed entity syntax which was the issue that @marot originally found which I was doing while travelling.

I've got some other changes coming, so will have a think about whether "just doing it" is better than throwing a helpful error.

Either way, a helpful error is better than the horrible errors users currently get, so would it be worth merging this as is?